noir-projects/noir-protocol-circuits/crates/types/src/hash.nr

```diff
@@ -193,32 +198,29 @@ pub fn accumulate_sha256(input: [U128; 4]) -> [Field; NUM_FIELDS_PER_SHA256] {
     // only occupies 128 bits.
     //
     // TODO(David): This does not seem to be getting guaranteed anywhere in the code?
-    //
-    // Concatenate 4 u128 bit integers into a byte array.
-    let mut hash_input_flattened = [0; 64];
-    for offset in 0..4 {
-        let input_as_bytes = input[offset].to_be_bytes();
-        for byte_index in 0..16 {
-            hash_input_flattened[offset * 16 + byte_index] = input_as_bytes[byte_index];
+
+    // Concatentate two fields into 31x2 = 62 bytes
+    // accumulate_sha256 assumes that the inputs are pre-truncated 31 byte numbers
+    let mut hash_input_flattened = [0; 62];
+    for offset in 0..NUM_FIELDS_PER_SHA256 * 2 {
```

Comment:
Can be `0..input.len()` so that we don't have to update it when we change the way it's calculated.